rfps/ethereum-test-vectors.md

```diff
@@ -0,0 +1,101 @@
+# RFP: Tool for applying test vectors from Ethereum on FEVM
+
+## Project Description
+
+FVM 2.1 which introduces the support for Ethereum smart contracts in the Filecoin network (FEVM) will be the biggest network upgrade since the mainnet launch. At the same time, it immensly widens the area of possible problems. That’s why it must be tested thoroughly.
+One of possibilities to test FEVM is to run already existing and proven smart contracts from Ethereum and compare the state it produces on Ethereum vs Filecoin.
+
+We are seeking proposals from teams that can build such a testing tool. This tool is expected to be in the form of a CLI.
+It should be able to:
+
+1. Fetch and store on the order of hundreds of contracts from Ethereum mainnet. These should be widely used contracts covering various use-case. 
+2. Deploy these contracts to the Filecoin testnet (called Butteflynet).
+3. Fetch thousands of transactions sent to these contracts on Ethereum in the past. 
+4. Replay these transactions to contracts deployed on Filecoin. 
+5. Compare the Ethereum state before and after these transactions with the results on Filecoin. We expect the team to research the ideas of how to do state comparison reliably.
+
+We want to run this tool a few times if needed and fix encountered errors every time.
+This tool can be written in either Go or Rust.
+
+## List of deliverables
+
+We expect the team to provide a Github repository containing
+* Source code of the tool
+* Guides on how to use it
+* Fetched smart contracts and transactions so the same data doesn’t have to be re fetched multiple times
+
+We also expect a demo of running all steps against the Wallaby (TBD) testnet to prove it works.
+
+## Timeline
+
+We expect this work to be done till 7th of November. We want to be able to use this tool after we deploy FVM M2.1 to Butterflynet. This is scheduled for 14th of November 2022.
```

Comment:
We expect this work to be done **until the** 7th of November. We would like to be able to use this tool after we deploy FVM M2.1 to Butterflynet, which is scheduled for **the** 14th of November 2022.